quality_of_service_demo/rclpy/quality_of_service_demo_py/incompatible_qos.py

```diff
@@ -0,0 +1,121 @@
+        'incompatibility', type=str,
+        help='Incompatibility type: durability, deadline, liveliness_policy, \
+                liveliness_lease_duration, reliability.')
+    return parser
+
+
+def main(args=None):
+    parser = get_parser()
+    parsed_args = parser.parse_args()
+    rclpy.init(args=args)
+
+    qos_profile_publisher = QoSProfile(depth=10)
+    qos_profile_subscription = QoSProfile(depth=10)
+
+    topic = 'incompatible_qos_chatter'
+    qos_policy_name = parsed_args.incompatibility
+
+    if qos_policy_name == 'durability':
+        print('Durability incompatibility selected.\n \
+Incompatibility condition: publisher durability kind < \
+subscripition durability kind.\n \
+Setting publisher durability to: VOLATILE\n \
+Setting subscription durability to: TRANSIENT_LOCAL\n')
+        qos_profile_publisher.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_VOLATILE
+        qos_profile_subscription.durability = \
+            QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL
+    elif qos_policy_name == 'deadline':
+        print('Deadline incompatibility selected.\n \
+Incompatibility condition: publisher deadline > subscription deadline.\n \
+Setting publisher durability to: 2 seconds\n \
+Setting subscription durability to: 1 second\n')
+        qos_profile_publisher.deadline = Duration(seconds=2)
+        qos_profile_subscription.deadline = Duration(seconds=1)
+    elif qos_policy_name == 'liveliness_policy':
+        print('Liveliness Policy incompatibility selected.\n \
+Incompatibility condition: publisher liveliness policy < \
+subscripition liveliness policy.\n \
+Setting publisher liveliness policy to: MANUAL_BY_NODE\n \
+Setting subscription liveliness policy to: MANUAL_BY_TOPIC\n')
+        qos_profile_publisher.liveliness = \
+            QoSLivelinessPolicy.RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE
+        qos_profile_subscription.liveliness = \
+            QoSLivelinessPolicy.RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC
+    elif qos_policy_name == 'liveliness_lease_duration':
+        print('Liveliness lease duration incompatibility selected.\n \
+Incompatibility condition: publisher liveliness lease duration > \
+subscription liveliness lease duration.\n \
+Setting publisher liveliness lease duration to: 2 seconds\n \
+Setting subscription liveliness lease duration to: 1 second\n')
+        qos_profile_publisher.liveliness_lease_duration = Duration(seconds=2)
+        qos_profile_subscription.liveliness_lease_duration = Duration(seconds=1)
+    elif qos_policy_name == 'reliability':
+        print('Reliability incompatibility selected.\n \
+Incompatibility condition: publisher reliability < subscripition reliability.\n \
+Setting publisher reliability to: BEST_EFFORT\n \
+Setting subscription reliability to: RELIABLE\n')
+        qos_profile_publisher.reliability = \
+            QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT
+        qos_profile_subscription.reliability = \
```

Comment:
Are the `\` needed here?

Reply:
For long assignments it does look like it is needed, AFAIK. 

Another thing that can be done is to assign one of the variables a shorter name, like [this](https://stackoverflow.com/questions/29555608/break-a-long-assignment-into-two-lines-in-python)

And I think you can also put the R.H.S inside parenthesis.